src/updater.cc

```diff
@@ -42,7 +44,12 @@ PollingMetadataUpdater::~PollingMetadataUpdater() {
   }
 }
 
-void PollingMetadataUpdater::start() {
+bool PollingMetadataUpdater::start() {
+  if (!reader_->ValidateConfiguration()) {
```

Comment:
Let's just make `ValidateConfiguration` a virtual method in `PollingMetadataUpdater`.

Reply:
Will address as a part of the conversation for the `reader` above.